src/engine/LocalVocab.h

```diff
@@ -91,6 +91,17 @@ class LocalVocab {
   // of the `vocabs`. The primary word set of the newly created vocab is empty.
   static LocalVocab merge(std::span<const LocalVocab*> vocabs);
 
+  // Merge all passed local vocabs to keep alive all the words from each of the
+  // `vocabs`.
+  template <std::ranges::range R>
+  void mergeWith(R&& vocabs) {
+    auto inserter = std::back_inserter(otherWordSets_);
+    for (const auto& vocab : AD_FWD(vocabs)) {
```

Comment:
Can be `const R& vocabs` and without `AD_FWD`.

Reply:
All the merge/share routines could skip all empty local vocabs. 
That is an easy optimization.

Reply:
> All the merge/share routines could skip all empty local vocabs. That is an easy optimization.

Does it though? To determine the size it does iterate over the same data structures just to calculate the size